src/world/FieldMessage.cpp

```diff
@@ -6,214 +6,139 @@
 namespace rtt {
 
 FieldMessage::FieldMessage(proto::SSL_GeometryFieldSize sslFieldSize) {
-  length = mm_to_m(sslFieldSize.field_length());
-  width = mm_to_m(sslFieldSize.field_width());
-  goalWidth = mm_to_m(sslFieldSize.goal_width());
-  goalDepth = mm_to_m(sslFieldSize.goal_depth());
-  boundaryWidth = mm_to_m(sslFieldSize.boundary_width());
-
-  for (proto::SSL_FieldLineSegment line : sslFieldSize.field_lines()) {
-    FieldLineSegment newLine;
-    newLine.name = std::string(name_map[line.name()]);
-    newLine.begin = mm_to_m(line.p1());
-    newLine.end = mm_to_m(line.p2());
-    newLine.thickness = mm_to_m(line.thickness());
-
-    addHelperLine(newLine);
-    field_lines.push_back(newLine);
-  }
-  
-  for (proto::SSL_FieldCicularArc arc : sslFieldSize.field_arcs()) {
-    FieldArc newArc;
-    newArc.name = std::string(name_map[arc.name()]);
-    newArc.center = mm_to_m(arc.center());
-    newArc.a1 = mm_to_m(arc.a1());
-    newArc.a2 = mm_to_m(arc.a2());
-    newArc.radius = mm_to_m(arc.radius());
-    newArc.thickness = mm_to_m(arc.thickness());
-
-    addHelperArc(newArc);
-    field_arcs.push_back(newArc);
-  }
-
+    initFieldLines(sslFieldSize);
+    initFieldArcs(sslFieldSize);
+    initFieldValues(sslFieldSize);
+    initFieldVectors();
+}
+
+void FieldMessage::initFieldValues(const proto::SSL_GeometryFieldSize &sslFieldSize) {
+    fieldValues[FIELD_LENGTH] = mm_to_m(sslFieldSize.field_length());
+    fieldValues[FIELD_WIDTH] = mm_to_m(sslFieldSize.field_width());
+    fieldValues[GOAL_WIDTH] = mm_to_m(sslFieldSize.goal_width());
+    fieldValues[GOAL_DEPTH] = mm_to_m(sslFieldSize.goal_depth());
+    fieldValues[BOUNDARY_WIDTH] = mm_to_m(sslFieldSize.boundary_width());
+    fieldValues[LEFTMOST_X] = fieldLines[LEFT_LINE].begin.x;
+    fieldValues[RIGHTMOST_X] = fieldLines[RIGHT_LINE].begin.x;
+    fieldValues[BOTTOMMOST_Y] = fieldLines[BOTTOM_LINE].begin.y;
+    fieldValues[TOPMOST_Y] = fieldLines[TOP_LINE].begin.y;
+    fieldValues[CENTER_Y] = fieldLines[CENTER_LINE].begin.y;
+}
```

Comment:
For the fieldValues[LEFTMOST_X] and so on I fixed this. But for the fieldValues[BOUNDARY_WIDTH] and before I did not change it, because in the earlier version of the FieldMessage class it would have been a problem as well.